packages/reporter/src/commands/command-model.ts

```diff
@@ -86,12 +90,16 @@ export default class Command extends Instrument {
 
     return this._isOpen || (this._isOpen === null
       && (
-        (this.group && this.type === 'system' && this.hasChildren) ||
-        (this.hasChildren && !this.event && this.type !== 'system') ||
-        _.some(this.children, (v) => v.hasChildren) ||
+        // command has nested commands
+        (this.name !== 'session' && this.hasChildren && !this.event && this.type !== 'system') ||
+        // command has nested commands with children
+        (this.name !== 'session' && _.some(this.children, (v) => v.hasChildren)) ||
+        // last nested command is open
         _.last(this.children)?.isOpen ||
+        // show slow command when test is running
         (_.some(this.children, (v) => v.isLongRunning) && _.last(this.children)?.state === 'pending') ||
-        _.some(this.children, (v) => v.state === 'failed')
+        // at last nested command failed
+        _.last(this.children)?.state === 'failed'
```

Comment:
The rules for whether or not a command is open is getting really complex - not saying we can really do much right here right now, but I hope we can move this to a more data driven fashion at some point. It feels like there's some kind of rule that can be generalized/extracted from all these conditionals. I'm not sure what it is, though.

Reply:
Yeah I agree, these rules are fairly complex. Currently these attributes are all event driven from the driver to the reporter so as attributes are updated, it's reflected here in the model. I'm open to all suggestions!